packages/core/utils/src/common/read-dir-recursive.ts

```diff
@@ -2,21 +2,51 @@ import { Dirent } from "fs"
 import { readdir } from "fs/promises"
 import { join } from "path"
 
-export async function readDirRecursive(dir: string): Promise<Dirent[]> {
-  let allEntries: Dirent[] = []
-  const readRecursive = async (dir) => {
+const MISSING_NODE_ERRORS = ["ENOTDIR", "ENOENT"]
+
+export async function readDir(
+  dir: string,
+  options?: {
+    ignoreMissing?: boolean
+  }
+) {
+  try {
     const entries = await readdir(dir, { withFileTypes: true })
+    return entries
+  } catch (error) {
+    if (options?.ignoreMissing && MISSING_NODE_ERRORS.includes(error.code)) {
+      return []
+    }
+    throw error
+  }
+}
 
-    for (const entry of entries) {
-      const fullPath = join(dir, entry.name)
-      Object.defineProperty(entry, "path", {
-        value: dir,
-      })
-      allEntries.push(entry)
+export async function readDirRecursive(
```

Comment:
NOTE: Is there any difference with the native `readdir` function with the recursive options set to true? If not then maybe we can eventually switch to the netive impl?

Reply:
There are differences from Node version to Node version. That's why I created the helper function. The loader was breaking in some versions